daemon/algod/api/client/restClient.go

```diff
@@ -572,23 +584,48 @@ func (client RestClient) GetGoRoutines(ctx context.Context) (goRoutines string,
 	return
 }
 
+type compileParams struct {
+	SourceMap bool `url:"sourcemap,omitempty"`
+}
+
 // Compile compiles the given program and returned the compiled program
-func (client RestClient) Compile(program []byte) (compiledProgram []byte, programHash crypto.Digest, err error) {
+func (client RestClient) Compile(program []byte, useSourceMap bool) (compiledProgram []byte, programHash crypto.Digest, sourceMap *logic.SourceMap, err error) {
 	var compileResponse model.CompileResponse
-	err = client.submitForm(&compileResponse, "/v2/teal/compile", program, "POST", false, true, false)
+
+	compileRequest := compileParams{SourceMap: useSourceMap}
+
+	err = client.submitForm(&compileResponse, "/v2/teal/compile", compileRequest, program, "POST", false, true, false)
 	if err != nil {
-		return nil, crypto.Digest{}, err
+		return nil, crypto.Digest{}, nil, err
 	}
 	compiledProgram, err = base64.StdEncoding.DecodeString(compileResponse.Result)
 	if err != nil {
-		return nil, crypto.Digest{}, err
+		return nil, crypto.Digest{}, nil, err
 	}
 	var progAddr basics.Address
 	progAddr, err = basics.UnmarshalChecksumAddress(compileResponse.Hash)
 	if err != nil {
-		return nil, crypto.Digest{}, err
+		return nil, crypto.Digest{}, nil, err
 	}
 	programHash = crypto.Digest(progAddr)
+
+	existenceOfSourceMap := compileResponse.Sourcemap != nil
+
+	if existenceOfSourceMap != useSourceMap {
+		return nil, crypto.Digest{}, nil, fmt.Errorf("useSourceMap arg %v not agreeing with existence of source-map %v", useSourceMap, existenceOfSourceMap)
```

Comment:
it's an error if I don't want to use a sourcemap that I've been given?

Reply:
my bad, I think something is off with this part of logic, probably need to rewrite a bit.

Reply:
see [e788b1d](https://github.com/algorand/go-algorand/pull/5253/commits/e788b1da26c25fc0ef13ca0157dce09522b8319c)